fastapi_sqla/sqla.py

```diff
@@ -50,7 +70,12 @@ def startup(key: str = _DEFAULT_SESSION_KEY):
         raise
 
     Base.prepare(engine)
-    _session_factories[key] = sessionmaker(bind=engine)
+
+    if has_sqlmodel:
+        _session_factories[key] = sessionmaker(bind=engine, class_=SQLModelSession)
```

Comment:
Nit, but since we determine whether SQLModel is supported at import time, would it not be a little less indirected to assign the appropriate session class at same time as import?

```py

try:
    from sqlmodel import Session as SessionCls
except ImportError:
    from sqlalchemy.orm.session import Session as SessionCls
```

And IIUC this should still work with typings since type checkers actually import + interpret modules.

Reply:
Done, I kept using `SqlaSession` to avoid breaking change (even though SQLASession would be better)